### ISSUE TITLE
In-Game: Crash when draw_sprite_ext() draws a Spine sprite with sprite_index 0 inside a with() on a struct

### ISSUE THREAD

**Issue by BscotchSeth** [YoYoGames/GameMaker-Bugs#4879]
### Description

Normally, you can draw sprites within a struct as long as you are currently within a `draw` event of an instance. For example, you can do this within an instance's draw event:

```
var _struct = {};
with _struct {
    draw_sprite_ext(sp_something, 0, 0, 0);
}
```

However, if you attempt to do this with a sprite that is a Spine sprite **and** that sprite has a sprite index of 0, you will get this crash:

```
ERROR!!! :: ############################################################################################
ERROR in
action number 1
of Draw Event
for object Object2:


Attempt to dispatch event on non-instance object
 at gml_Object_Object2_Draw_0 (line 7) - 	draw_sprite_ext(Sprite1, 0, 0, 0, 1, 1, 0, c_white, 1);
############################################################################################
gml_Object_Object2_Draw_0 (line 7)
```

This crash doesn't occur if it's a non-Spine sprite whose sprite index is 0, and it also doesn't occur if it's a Spine sprite whose sprite index is not 0. I have attached a Contact Us package with a test project that demonstrates the issue.

[SpriteZeroCrash.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/14488021/SpriteZeroCrash.zip)

### Expected Change

Sprites shouldn't cause crashes when you draw them, regardless of their sprite index.

### Steps To Reproduce

1. Open the test project
2. Run it
3. Feel pain in your soul as your world shatters and misery clouds your mind
4. Observe the error log

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.200 (Betas)

### Which platform(s) are you seeing the problem on?

Windows

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by shichen85** [comment]
FYI, in case the project has several sprites, either
```
    var s0 = handle_parse("ref sprite 0");
    draw_sprite_ext(s0, 0, 0, 0, 1, 1, 0, c_white, 1);
```
or just use the integer 0 as the argument:
```
draw_sprite_ext(0, 0, 0, 0, 1, 1, 0, c_white, 1);
```
Will point to the 0th indexed sprite.

**Comment by BscotchSeth** [comment]
Good tip! Although the 0th sprite still must be a Spine sprite for this crash to occur.

**Comment by rwkay** [comment]
Fixed in 2024.4
